### PR TITLE
[CORE-194] Adding Unit Tests to UploadData Component

### PR DESCRIPTION
**Pull request by kevinmarete**
### What
- This PR adds unit tests for the `UploadData` component to ensure it works as expected.

### Why
- Improves code quality by ensuring the UploadData component is robust and bug-free.

### Testing strategy
- Running Unit Tests

### CODE REVIEW

**Review by calypsomatic — APPROVED** [review]

File: src/workspace-data/upload-data/UploadData.test.tsx [hunk 1]
```diff
@@ -77,23 +57,70 @@ jest.mock('src/libs/feature-previews', () => ({
...
 
+  const uploadDataFileToStorage = async () => {
+    const uploadBtn = screen.getByText('Upload');
+
+    await expect(uploadBtn).toBeInTheDocument();
+
+    fireEvent.click(uploadBtn);
+  };
+
   it('Renders data uploader', () => {
```
Comment (msilva-broad) [inline src/workspace-data/upload-data/UploadData.test.tsx:120]: please add standard // Arrange, // Act, and // Assert sections to these tests to improve readability. (blank line in between sections).  For some tests that are using helpers, you can label first section as // Arrange/Act

**Comment by msilva-broad** [comment]
great solid start to some unit test coverage here.  Nice work.

File: src/workspace-data/upload-data/UploadData.test.tsx (hunk 1 above)
Comment (kevinmarete) [reply src/workspace-data/upload-data/UploadData.test.tsx:120]: Done: https://github.com/DataBiosphere/terra-ui/pull/5181/commits/b36edc7b6e16eb06a6bcc458f8f56b7ee87e4919

File: src/workspace-data/upload-data/UploadData.test.tsx [hunk 2]
```diff
@@ -73,65 +53,125 @@ jest.mock('src/workspaces/common/state/useWorkspaces', () => ({
...
+    // Arrange
     renderWithAppContexts(<UploadData />);
+
+    // Act & Assert
```
Comment (msilva-broad) [inline src/workspace-data/upload-data/UploadData.test.tsx]: this block of tests look like they should all have the last section just labeled as // Assert, and you can either add Act to the // Arrange comment, or leave it out.

Comment (kevinmarete) [reply src/workspace-data/upload-data/UploadData.test.tsx]: Updated: https://github.com/DataBiosphere/terra-ui/pull/5181/commits/e3ef0c7a04fabe0c854833a0389d4b597c36231b

**Review by msilva-broad — APPROVED** [review]
nice work!

**Pull request merged**